main/updater.js

```diff
@@ -34,7 +37,9 @@ function beforeQuitCleanup () {
   app.removeAllListeners('window-all-closed')
 }
 
-function setup (/** @type {import('./typings').Context} */ ctx) {
+async function setup (/** @type {import('./typings').Context} */ ctx) {
```

Comment:
If we change setup to an async function, then I think we should also modify the code calling this function to await the outcome, right?

How about moving the `await` part into `onUpdateAvailable`?

```js
// at the top
const pRetryPromise = import ('p-retry').then(({default}) => (default))

// in `onUpdateAvailable`
pRetryPromise
.then(pRetry => pRetry(
   () => autoUpdater.downloadUpdate(),
   // etc.
)
.then(
    _ => log.info('Update downloaded'),
    err => log.error('Cannot download the update.', err)
  )
```

Reply:
This function is called in https://github.com/filecoin-station/desktop/blob/866e0887e4548189004fcb099bc3ef040f3a63d8/main/index.js#L173, where it's already being awaited. Or am I missing something?

Reply:
Ah, that's great to know. Thanks for checking. All is good; my comment can be ignored.